File: leetcode/0045-jump-game-ii/solution.c
int max(int a, int b) {
    return a>b?a:b;
}
int jump(int* nums, int n) {
    int jump = 0;
    int distance = 0;
    int next = 0;
    for(int i = 0;i<n;i++) {
        if(i > distance) {
            distance = next;
            jump++;
        }
        next = max(next, i+nums[i]);
    }
    return jump;
}

File: leetcode/0045-jump-game-ii/solution.rs
impl Solution {
    pub fn jump(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut jump = 0;
        let mut distance = 0;
        let mut next = 0;
        for i in 0..n {
            if i > distance {
                jump += 1;
                distance = next;
            }
            next = std::cmp::max(next, i+nums[i] as usize);
        }
        return jump;
    }
}
